### ISSUE TITLE
Using @ts-expect-error in a @ts-nocheck file

### ISSUE THREAD

**Issue by nicojs** [microsoft/TypeScript#39427]
**TypeScript Version:**  3.9 - 4.0.0-beta, v4.0.0-dev.20200705 

**Search Terms:** "ts-nocheck ts-expect-error" "ts-expect-error"

**Code**

```ts
// @ts-nocheck
// @ts-expect-error
'a' - 'b'
```

**Expected behavior:** No compile error

**Actual behavior:** Error: Unused '@ts-expect-error' directive.(2578)

**Playground Link:** https://www.typescriptlang.org/play?ts=4.0.0-beta#code/PTAEAEBcGcFoDsD2BjAFgU2QawFAgjLOgB4AOmkRATlYlTgOQCGDosoDARg0A

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
![image](https://user-images.githubusercontent.com/6685088/86851848-83e24f80-c068-11ea-8b16-3eefabcd3a84.png)

**Comment by JoshuaKGoldberg** [comment]
What direction does this bug point @RyanCavanaugh? I would think that the code is behaving as expected: the `// @ts-expect-error` is redundant and shouldn't exist. Perhaps a more specific error message?

```
Error: Unused '@ts-expect-error' directive because of an earlier '@ts-no-check' directive.
  '@ts-no-check' was specified here: ...
```

Per the comments in https://github.com/stryker-mutator/stryker/issues/2403#issuecomment-677519240 I think the right resolution would be for fancy code mutation services to strip out comments. cc @nicojs, since it looks like you already implemented that in https://github.com/stryker-mutator/stryker/pull/2365, right?

**Comment by JoshuaKGoldberg** [comment]
What direction does this bug point @RyanCavanaugh? I would think that the code is behaving as expected: the `// @ts-expect-error` is redundant and shouldn't exist. Perhaps a more specific error message?

```
Error: Unused '@ts-expect-error' directive because of an earlier '@ts-no-check' directive.
    '@ts-no-check' was specified here: ...
```

Per the comments in https://github.com/stryker-mutator/stryker/issues/2403#issuecomment-677519240 I think the right resolution would be for fancy code mutation services to strip out comments. cc @nicojs, since it looks like you already implemented that in https://github.com/stryker-mutator/stryker/pull/2365, right?

**Comment by RyanCavanaugh** [comment]
I think `nocheck` is the higher-order bit here; the intent of that is to "shut up for the whole file" and that should include `expect-error` not issuing an error.

**Comment by JoshuaKGoldberg** [comment]
This actually doesn't give an error anymore in 4.1.5 and up: [original example](https://www.typescriptlang.org/play?ts=4.1.5#code/PTAEAEBcGcFoDsD2BjAFgU2QawFAgjLOgB4AOmkRATlYlTgOQCGDosoDARg0A); [example with code between the two comments](https://www.typescriptlang.org/play?ts=4.1.5&ssl=1&ssc=1&pln=5&pc=13#code/PTAEAEBcGcFoDsD2BjAFgU2QawFAEZQBqUAJgG4ccQIZZ0APAB00joCc3E2qwByAQ16hYoXgCNeQA). 🤷
